### PR TITLE
Improve SDK concepts (screencaps, diagrams, etc)

### PR DESCRIPTION
**Pull request by alexandratran**
- Update SDK screen recordings and linked examples - [preview](https://docs.metamask.io/sdk-concepts/wallet/concepts/sdk/)
- Add SDK connections page with diagrams using Mermaid plugin - [preview](https://docs.metamask.io/sdk-concepts/wallet/concepts/sdk/connections/)
  - (I couldn't edit the theme/colors of the diagrams while making both light/dark mode work, so I stuck with the default theme.)
- Add redirect `docs.metamask.io/sdk` -> SDK setup page
  - test redirect on the PR branch: https://docs.metamask.io/sdk-concepts/sdk
- Other minor edits to related content

### CODE REVIEW

**Review by andreahaku — APPROVED** [review]
LGTM

File: wallet/concepts/sdk/android.md
Comment (mjsmike62) [inline wallet/concepts/sdk/android.md:1]: Should the following be an unordered list...

1. The dapp, with the SDK installed, initiates communication when a user connects to MetaMask.
    The dapp deeplinks to MetaMask, and MetaMask sets up the Android Native Module to receive client requests.
2. The dapp generates an ECIES public/private key pair.
    The dapp and MetaMask exchange public keys over IPC.

File: wallet/concepts/sdk/connections.md
Comment (mjsmike62) [inline wallet/concepts/sdk/connections.md:1]: code/deeplink = code or deeplink ?

File: wallet/concepts/sdk/index.md
Comment (mjsmike62) [inline wallet/concepts/sdk/index.md:1]: third party = third-party

Edit for all occurrences. 

For... It also supports deeplinking on mobile platforms, such as in the following screen recording.

Not sure if I understand what is trying to be said. Could "such as in the following screen recording" be replaced with "as demonstrated in the following screen recording."?

**Review by mjsmike62 — APPROVED** [review]

File: wallet/concepts/sdk/android.md
Comment (alexandratran) [reply wallet/concepts/sdk/android.md:1]: It's a sequence of steps in the flow corresponding to the diagram, so ordered list makes more sense IMO.

**Pull request merged**